src/filters/filters.jl

```diff
@@ -106,8 +234,75 @@ function Base.append!(filters::FilterPipeline, extra)
     end
     return filters
 end
+function Base.push!(p::FilterPipeline, f::F) where F <: Filter
+    ref = Ref(f)
+    GC.@preserve ref begin
+        API.h5p_set_filter(p.plist, filterid(F), API.H5Z_FLAG_OPTIONAL, div(sizeof(F), sizeof(Cuint)), pointer_from_objref(ref))
+    end
+    return p
+end
+function Base.push!(p::FilterPipeline, f::UnknownFilter)
+    GC.@preserve f begin
+        API.h5p_set_filter(p.plist, f.filter_id, f.flags, length(f.data), pointer(f.data))
+    end
+end
+
+# Generic implementation of register_filter
+"""
+    register_filter(filter::F) where F <: Filter
+
+Register the filter with the HDF5 library via API.h5z_register.
+Also add F to the FILTERS dictionary.
+"""
+function register_filter(filter::F) where F <: Filter
```

Comment:
Is there any reason this needs to be defined on an instance rather than just a type?